src/masternodes/masternodes.cpp

```diff
@@ -618,17 +620,34 @@ bool CRewardsHistoryStorage::Flush()
     return CCustomCSView::Flush();
 }
 
-isminetype IsMineCached(CWallet const & wallet, CScript const & script)
+struct CCacheInfo
```

Comment:
what are these changes for? is it an optimisation? will this speed up indexing too?

Reply:
I guess the change is to dynamically update the cache if the wallet changed or unloaded.

Reply:
Update mine cache when you insert private key, also clears cache when wallet is unloaded.